autopilot/agent.go

```diff
@@ -589,7 +609,8 @@ func (a *Agent) controller() {
 						return
```

Comment:
Here, and also in the `alreadyConnected` return above, shouldn't the node be deleted from `pendingConns`? Otherwise it will not be retried ever again, even though new channels may need to be opened in the future.

Reply:
Yes you're right, I missed that case. Fixed!

Reply:
Now all the flows following `ConnectToPeer`, start with obtaining the lock and removing the pending connection (if the pending opens set check fails, it is even removed twice now). I think you could lift locking and removing up and do it right after `ConnectToPeer`.

Reply:
I think there may be some cases where we want to delay removing from `pendingConns`. If the entry is removed right after connecting peers, we may later call `DisconnectPeer`. Since disconnecting will cancel all known active/pending connections, this may interfere with any subsequent attachment directives that are trying to connect to the same peer.